pkg/registry/storage/volumeattachment/strategy.go

```diff
@@ -139,4 +141,13 @@ func (volumeAttachmentStatusStrategy) PrepareForUpdate(ctx context.Context, obj,
 
 	newVolumeAttachment.Spec = oldVolumeAttachment.Spec
 	metav1.ResetObjectMetaForStatus(&newVolumeAttachment.ObjectMeta, &oldVolumeAttachment.ObjectMeta)
+
+	if !feature.DefaultFeatureGate.Enabled(features.MutableCSINodeAllocatableCount) {
+		if newVolumeAttachment.Status.AttachError != nil {
```

Comment:
The field should only be cleared if it's not already set in the old object.

Reply:
Done in latest rev